src/puppetlabs/puppetdb/import.clj

```diff
@@ -12,12 +12,36 @@
             [puppetlabs.puppetdb.export :as export]
             [puppetlabs.puppetdb.utils :as utils]
             [puppetlabs.puppetdb.cheshire :as json]
-            [puppetlabs.puppetdb.schema :refer [defn-validated]]))
+            [puppetlabs.puppetdb.schema :refer [defn-validated]]
+            [puppetlabs.puppetdb.utils :as utils]))
 
 (defn file-pattern
   [entity]
   (re-pattern (str "^" (.getPath (io/file utils/export-root-dir entity ".*\\.json")) "$")))
 
+(def after-certname-length
+  "The format is <certname>-<date hash><suffix>. This is a count of
+  the characters after the certname"
+  (+ 1
+     40
+     (count export/file-suffix)))
+
+(def export-root-dir-length
+  (count utils/export-root-dir))
+
+(defn extract-certname [^String command-type ^String path]
+  (let [path-length (.length path)
+        prefix-length (+ export-root-dir-length
+                         (.length command-type)
+                         2)]
+    (subs path prefix-length (- path-length after-certname-length))))
+
+(defn extract-facts-certname [^String path]
+  (let [prefix-length (+ export-root-dir-length
+                         (.length "facts")
+                         2)]
+    (subs path prefix-length (- (.length path) (count export/file-suffix)))))
```

Comment:
Interesting, we just had a discussion around this, another vote in the favor of a regex here is I think we can use it here and instead of the file-pattern from line 18 above. I'll try switching it over and putting some tests around it.